modules/build/src/main/scala/scala/build/bsp/BspServer.scala

```diff
@@ -137,6 +137,19 @@ class BspServer(
       sourceItem.setUri(updatedUri)
       sourceItem.setGenerated(false)
     }
+
+    // GeneratedSources not corresponding to files that exist on disk (unlike script wrappers)
+    val sourcesWithReportingPathString = generatedSources.values.flatMap(_.sources)
+      .filter(_.reportingPath.isLeft)
+
+    for {
+      item <- res.getItems.asScala
+      if validTarget(item.getTarget)
+      sourceItem <- item.getSources.asScala
+      if sourcesWithReportingPathString.exists(
+        _.generated.toNIO.toUri.toASCIIString == sourceItem.getUri
+      )
+    } sourceItem.setGenerated(true)
```

Comment:
I was confused regarding to the type of `reportingPath`. I think it shouldn't be an `Either` type but rather, it could be better represented by a case class with Scala doc explaining the type of path.

But I think It can be added in the new following PR.

Reply:
Can you create an issue to ensure we don't forget about this?

Reply:
#2308